### PR TITLE
ironic: Store configdrive in Swift if possible

### PR DESCRIPTION
**Pull request by skazi0**
Allow storage of configdrive in Swift if it is available.
If Swift is available but user doesn't want to store the configdrive there it can be disabled using added attribute. It's value is ignored if Swift is not available.

Additional commit fixes username/password fields in swift settings.

### CODE REVIEW

File: chef/cookbooks/ironic/libraries/helpers.rb [hunk 1]
```diff
@@ -62,6 +62,8 @@ def swift_settings(node, glance)
         glance_container: glance[:glance][:swift][:store_container],
         glance_account: glance_account,
         protocol: swift[:swift][:ssl][:enabled] ? "https" : "http",
+        service_user: swift[:swift][:service_user],
+        service_password: swift[:swift][:service_password],
```
Comment (vuntz) [inline chef/cookbooks/ironic/libraries/helpers.rb]: Are these really used?

Comment (skazi0) [reply chef/cookbooks/ironic/libraries/helpers.rb]: Here they are:
https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/ironic/templates/default/ironic.conf.erb#L45

Comment (skazi0) [reply chef/cookbooks/ironic/libraries/helpers.rb]: The template might drop the conditions but I'm not sure why they were added there.

Comment (vuntz) [reply chef/cookbooks/ironic/libraries/helpers.rb]: Thanks, that wasn't obvious to me.

**Pull request merged**